### PR TITLE
Psd 777 pact

### PR DESCRIPTION
**Pull request by FayToward**
https://eaflood.atlassian.net/browse/PSD-777

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3,6 +3,7 @@ library("defra-library@$BRANCH_NAME")
 def pr = ''
 def repoName = ''
 def versionFileName = 'VERSION'
+def postTestTasks = {
```
Comment (johnwatson484) [inline Jenkinsfile:6]: Don't need to make any changes to this file.

File: vars/buildDotNetCore.groovy [hunk 2]
```diff
@@ -3,6 +3,8 @@ def call(Map config=[:]) {
   def mergedPrNo = ''
   def pr = ''
   def repoName = ''
+  def version = version.getPackageJsonVersion()
+  def commitSha = utils.getCommitSha()
```
Comment (johnwatson484) [inline vars/buildDotNetCore.groovy:7]: Don't need to do this as already covered in the getVariables function.

File: vars/buildNodeJs.groovy [hunk 3]
```diff
@@ -7,6 +7,8 @@ def call(Map config=[:]) {
   def pr = ''
   def tag = ''
   def mergedPrNo = ''
+  def version = version.getPackageJsonVersion()
+  def commitSha = utils.getCommitSha()
```
Comment (johnwatson484) [inline vars/buildNodeJs.groovy:11]: Same comments as .NET Core

File: Jenkinsfile (hunk 1 above)
Comment (FayToward) [reply Jenkinsfile:6]: i'm not convinced i did but i shall remove it

File: vars/buildDotNetCore.groovy [hunk 4]
```diff
@@ -51,6 +54,10 @@ def call(Map config=[:]) {
         build.runTests(repoName, repoName, BUILD_NUMBER, tag)
       }
 
+     stage('Publish pact broker') {
+        pact.pacts(repoName, pact.string, pact.usernamePassword)
```
Comment (johnwatson484) [inline vars/buildDotNetCore.groovy:58]: Can probably reduce this to just `repoName` and have the class itself encapsulate all the logic

File: vars/pact.groovy [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+def string(credentialsId,variable){
+  Pact.string(this, credentialsId, variable)
+}
+
+def usernamePassword(credentialsId, usernameVariable, passwordVariable){
+  Pact.usernamePassword(this, credentialsId, usernameVariable, passwordVariable)
+}
+
+def pacts(repoName, string, usernamePassword){
```
Comment (johnwatson484) [inline vars/pact.groovy:11]: Maybe make this more descriptive, function names should be a verb.  eg publishPacts.

**Review by johnwatson484 — APPROVED** [review]
LGTM

**Pull request merged**